src/Microsoft.ML.Core/Utilities/Hashing.cs

```diff
@@ -125,23 +127,21 @@ public static uint MurmurRound(uint hash, uint chunk)
         /// * 0x0800 to 0xFFFF : 1110xxxx 10xxxxxx 10xxxxxx
         /// NOTE: This MUST match the StringBuilder version below.
         /// </summary>
-        public static uint MurmurHash(uint hash, string data, int ichMin, int ichLim, bool toUpper = false)
+        public static uint MurmurHash(uint hash, ReadOnlyMemory<char> data, bool toUpper = false)
```

Comment:
Since we don't need to store the `ReadOnlyMemory<char>`, this should be a `ReadOnlySpan<char>` instead.